lib/ansible/modules/cloud/vmware/vmware_portgroup_facts.py

```diff
@@ -98,54 +108,110 @@
 
 
 class PortgroupFactsManager(PyVmomi):
+    """Class to manage Port Group facts"""
     def __init__(self, module):
         super(PortgroupFactsManager, self).__init__(module)
         cluster_name = self.params.get('cluster_name', None)
         esxi_host_name = self.params.get('esxi_hostname', None)
         self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.policies = self.params.get('policies')
 
     @staticmethod
-    def normalize_pg_info(portgroup_obj):
+    def normalize_pg_info(portgroup_obj, policy_facts):
+        """Create Port Group information"""
         pg_info_dict = dict()
-        pg_info_dict['name'] = portgroup_obj.spec.name
-        vlan_id = 'N/A'
-        if portgroup_obj.spec.vlanId:
-            vlan_id = portgroup_obj.spec.vlanId
-        pg_info_dict['vlan_id'] = vlan_id
-        switch_name = 'N/A'
-        if portgroup_obj.spec.vswitchName:
-            switch_name = portgroup_obj.spec.vswitchName
-        pg_info_dict['vswitch_name'] = switch_name
-
-        # Network Policy related facts
-        pg_info_dict['promiscuous_mode'] = bool(portgroup_obj.spec.policy.security.allowPromiscuous)
-        pg_info_dict['mac_changes'] = bool(portgroup_obj.spec.policy.security.macChanges)
-        pg_info_dict['forged_transmits'] = bool(portgroup_obj.spec.policy.security.forgedTransmits)
+        spec = portgroup_obj.spec
+        pg_info_dict['portgroup'] = spec.name
+        pg_info_dict['vlan_id'] = spec.vlanId
```

Comment:
This will fail when there is no vlanId or vswitch in portgroup. I don't remember the exact reason, but I encountered error here sometimes in past.

Reply:
How can this fail? A portgroup must have a vswitch assigned and the vlan_id is either 0 (no tagging) or another ID

Reply:
Like I said I don't remember exactly where it happened but that is why I added that if block. I am ok without if block if it is working on your systems.

Reply:
It works without issues so far on 50+ ESXi hosts (all on 6.5 with the same build). If there's an issue then the `vmware_portgroup` module need to be changed as well. I would leave it like it is and later change both modules if there's an issue with those values in the spec.